index.js

```diff
@@ -1,77 +1,95 @@
+  }
+  else {
+    // For local storage, prepend the upload directory path when checking files
+    return callback(null, fs.existsSync(path.join(uploadsPath, filepath)));
+  }
+}
 
 app.use(express.json())
 app.get('/v1/obabel', (req, res) => {
   //identifier for this particular file conversion job
   let jobId = req.query.storage_hash;
-  //path defining the directory the file is to be saved to
-  let uploadsPath = path.join(__dirname,'uploads');
+  
   //the full path of the file to be saved
-  let fullPath = path.join(__dirname,"uploads", jobId);
-  //the full file path of the file to store openbabel's output text
-  let obabelOutputFilePath = path.join(fullPath,'obabel-output.txt');
+  let jobPath = path.join(jobId+"/");
+  
+  //the path of the zipped response file
+  let responsePath = useCloudStorage ? path.join(jobId,'response.zip') : jobId+'.zip';
+  
   //Keep track of whether a response has been sent to avoid sending a
   //redundant response
   let responseIsSent = false;
   
-  if (!fs.existsSync(fullPath)) {
-    res.status(400);
-    responseIsSent = true;
-    return res.send('No job with that ID.');
-    
-  }else if (!fs.existsSync(obabelOutputFilePath)) {
-    res.status(300);
-    responseIsSent = true;
-    return res.send('Job still processing.');
-      
-  } else {
+  checkExists(jobPath, (err, exists) => {
+    if (err) {
+      res.status(500);
+      responseIsSent = true;
+      return res.send(err);
+    }
+    else if (exists) {
+      checkExists(responsePath, (err, exists) => {
+        if (err) {
+          res.status(500);
+          responseIsSent = true;
+          return res.send(err);
+        }
+        else if (exists) {
+          try{
+            let output=null;
 
-    let outputPath = path.join(uploadsPath, jobId + '.zip');
-    let output = fs.createWriteStream(outputPath);
-    
-    let archive = archiver('zip', {
-      zlib: { level: 9 }
-    })
-    
-    output.on('close', function () {
-      let stat = fs.statSync(outputPath);
-      if(!responseIsSent){
-        res.writeHead(200, {
-          'Content-Type': 'application/zip',
-          'Content-Length': stat.size
-        });
-        let readStream = fs.createReadStream(outputPath);
-        readStream.pipe(res)
-      }
-    })
-    
-    archive.on('error', function (err) {
-      if(!responseIsSent){
-        res.status(500);
-        console.log(err)
-        responseIsSent = true;
-        return res.send('File archiving error.');
-      }
-    })
-    
-    archive.pipe(output)
-    console.log(jobId);
-    
-    try{
-      archive.directory(fullPath, "", { name: jobId })
-      archive.finalize();
-    } catch(error) {
-      console.log(error)
+            if (useCloudStorage) {
+              output = storage.bucket(bucket).file(responsePath).createReadStream();
+            }
+            else {
+              output = fs.createReadStream(path.join(uploadsPath, jobId+'.zip'));
```

Comment:
Technically a security risk if we leave the ability to create files on the server - could take up all the memory. We should have a way to delete the files if we keep this.

Reply:
Would still approve if you didn't fix this tho, since its out of our use case

Reply:
Yeah, given the use-case restrictions there, I'm gonna leave it be for now. Would add a `delete` endpoint if we had time, but it's not a huge deal.